### PR TITLE
Add tooltip info when tone equalizer is selected (like other modules)

### LINKED ISSUE #5669: FR: Have a way to disable modules from running on OpenCL
**Is your feature request related to a problem? Please describe.**
Considering issues like #3756, I feel that a global "turn off all OpenCL" is not too fine grained enough. I'm wondering if it is possible to blacklist certain problematic modules from running on OpenCL. 

**Describe the solution you'd like**
A button in the preset menu that has a check mark that lets you blacklist a module from running on OpenCL, falling back to the CPU implementation. This could also be used to communicate that there is no OpenCL implementation available.

It could also be done on a global module basis, not a module instance basis, by putting this info in the settings panel.

### LINKED ISSUE #5686: tone equalizer → add keyboard/mouse hints
Add keyboard/mouse hints (just like other modules do) on top of the image for tone equalizer module.

### PR DESCRIPTION
**Pull request by hatsunearu**
Fixes #5686. 

image

### CODE REVIEW

File: src/iop/toneequal.c [hunk 1]
```diff
@@ -1825,6 +1825,11 @@ static void switch_cursors(struct dt_iop_module_t *self)
     // if pipe is clean and idle and cursor is on preview,
     // hide GTK cursor because we display our custom one
     dt_control_change_cursor(GDK_BLANK_CURSOR);
+    dt_control_hinter_message(darktable.control,
+                  "scroll over image to change tone exposure by 0.25 EV\n"
+                  "shift+scroll to change by 1 EV, and ctrl+scroll to change by 0.1 EV\n"
+                  "A+scroll to zoom");
```
Comment (elstoc) [inline src/iop/toneequal.c:1831]: You can't hard-code "A+scroll" here. "A" is just the default key and can be changed in preferences.

File: src/iop/toneequal.c [hunk 2]
```diff
@@ -2278,6 +2284,14 @@ void gui_focus(struct dt_iop_module_t *self, gboolean in)
...
     dt_bauhaus_widget_set_quad_active(GTK_WIDGET(g->show_luminance_mask), FALSE);
     dt_dev_reprocess_center(self->dev);
+    dt_control_hinter_message(darktable.control, "");
+  }
+  else
+  {
+    dt_control_hinter_message(darktable.control,
+                              "scroll over image to change tone exposure by 0.25 EV\n"
+                              "shift+scroll to change by 1 EV, and ctrl+scroll to change by 0.1 EV\n"
+                              "A+scroll to zoom");
```
Comment (elstoc) [inline src/iop/toneequal.c:2294]: similarly here

Comment (hatsunearu) [reply src/iop/toneequal.c:2294]: Yikes. What should the text be? Should I try to read the preferences?

Comment (elstoc) [reply src/iop/toneequal.c:2294]: TBH I'd just remove this line. The same shortcut is also used to pan/zoom while editing masks and there probably isn't enough space to add it to the mask hinter messages as well. I think for some languages even the current mask hinter message doesn't completely fit in the available space.

Comment (elstoc) [reply src/iop/toneequal.c:2294]: If you really wanted to look it up I think it's stored in ``darktable.control->accels.darkroom_skip_mouse_events``

Comment (hatsunearu) [reply src/iop/toneequal.c]: Alright, I just removed the last line.

File: src/iop/toneequal.c [hunk 3]
```diff
@@ -1825,6 +1825,10 @@ static void switch_cursors(struct dt_iop_module_t *self)
     // if pipe is clean and idle and cursor is on preview,
     // hide GTK cursor because we display our custom one
     dt_control_change_cursor(GDK_BLANK_CURSOR);
+    dt_control_hinter_message(darktable.control,
+                              "scroll over image to change tone exposure by 0.25 EV\n"
```
Comment (TurboGit) [inline src/iop/toneequal.c:1829]: The numbers should not be there as this could change anytime.

Comment (TurboGit) [reply src/iop/toneequal.c:1829]: I would rewrite this to something like that:

```
scroll over image to change tone exposure
use shift+scroll  for larger steps
use ctrl+scroll for smaller steps.
```

File: src/iop/toneequal.c [hunk 4]
```diff
@@ -2278,6 +2282,13 @@ void gui_focus(struct dt_iop_module_t *self, gboolean in)
     g->mask_display = FALSE;
     dt_bauhaus_widget_set_quad_active(GTK_WIDGET(g->show_luminance_mask), FALSE);
     dt_dev_reprocess_center(self->dev);
+    dt_control_hinter_message(darktable.control, "");
+  }
+  else
+  {
+    dt_control_hinter_message(darktable.control,
+                              "scroll over image to change tone exposure by 0.25 EV\n"
```
Comment (TurboGit) [inline src/iop/toneequal.c:2290]: likewise

**Comment by AlicVB** [comment]
When you are at it, I think it would be great to also update the accel window (H shortcut)
For that, you "just" have to implement `mouse_actions` function. You can have an example here : https://github.com/darktable-org/darktable/blob/dca4c828b051c79bdb49869b8edf28993476833f/src/iop/clipping.c#L3500

This should be quite trivial, as you already have listed all the actions with their text... just ask if you need help !

**Comment by hatsunearu** [comment]
Hmm, I'm guessing I don't have to implement the A+scroll in that thing (since AFAIK that's a global keybind?)

**Comment by hatsunearu** [comment]
image

File: src/iop/toneequal.c (hunk 3 above)
Comment (hatsunearu) [reply src/iop/toneequal.c]: this is done.

**Comment by TurboGit** [comment]
Thanks !

**Pull request merged**